1306_expansion_statements/expansion-statements.md

```diff
@@ -469,6 +469,409 @@ void f() {
 
 ## Implementation experience
 
-TODO
+Bloomberg's Clang/P2996 fork (available [on Godbolt](https://godbolt.org/z/Yjv1dM4eK)) implements all features proposed by this paper. Expansion statements are enabled with the `-fexpansion-statements` flag (or with `-freflection-latest`).
 
 
+# Proposed wording
+
+## [basic.scope.pdecl] Point of declaration {-}
+
+Update paragraph 11 to specify the locus of an expansion statement:
+
+::: std
+[11]{.pnum} The locus of a `$for-range-declaration$` of a range-based `for` statement ([stmt.range]) is immediately after the `$for-range-initializer$`. [The locus of a `$for-range-declaration$` of an expansion statement ([stmt.expand]) is immediately after the `$expansion-initializer$`.]{.addu}
+
+:::
+
+## [basic.scope.block] Block scope {-}
+
+Update paragraph 1.1 to include expansion statements:
+
+::: std
+- [1.1]{.pnum} selection[,]{.addu} [or]{.rm} iteration[, or expansion]{.addu} statement ([stmt.select], [stmt.iter] [, [stmt.expand]]{.addu})
+
+:::
+
+## [class.temporary] Temporary objects {-}
+
+Modify paragraph 5 to clarify that there are now six contexts:
+
+::: std
+[5]{.pnum} There are [five]{.rm} [six]{.addu} contexts in which temporaries are destroyed at a different point than the end of the full-expression. [...]
+:::
+
+Insert a new paragraph after paragraph 7 to extend the lifetime of temporaries created by expansion statements, and update the ordinal number used in paragraph 8:
+
+::: std
+[7]{.pnum} The fourth context is when a temporary object other than a function parameter object is created in the `$for-range-initializer$` of a range-based `for` statement. [...]
+
+[[7+]{.pnum} The fifth context is when a temporary object other than a function parameter object is created in the `$expansion-initializer$` of an iterable or destructurable expansion statement, or in a full-expression in the `$expansion-init-list$` of an enumerated expansion statement ([stmt.expand]). If such a temporary object would otherwise be destroyed at the end of that full-expression, the object persists for the lifetime of the reference initialized by the expression in the expanded expansion statement.]{.addu}
```

Comment:
Why "other than a function parameter object"? What actually is a function parameter object?